core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -79,6 +88,22 @@ class DynamicConfigManager(private val zkUtils: ZkUtils,
                            private val time: Time = SystemTime) extends Logging {
   private var lastExecutedChange = -1L
 
+  // Apply all existing client/user configs to the ClientIdConfigHandler/UserConfigHandler to bootstrap the overrides
```

Comment:
The subscription to ZK change notification path was made in ZkNodeChangeNotificationListener. So, in order not to miss any changes, we should apply the bootstrap logic after ZkNodeChangeNotificationListener is initialized.